.changeset/warm-chicken-allow.md

```diff
@@ -0,0 +1,5 @@
+---
+'@sap-cloud-sdk/connectivity': minor
+---
+
+[Fixed Issue] Fix `getDestination()` to allow passing an async transform function for a service binding in the `options`. It also enables passing the transform function as a part of the `DestinationOrFetchOptions`, which is required by `execute()`, `executeHttpRequest()`, `executeRaw()`, etc.
```

Comment:
I know that the `transformFn` was marked as `@internal` but I would anyway write a compatibility note that the method was renamed and made `async`.

Reply:
I was a bit hesitating. I will add a compatibility note for this.